securebanking-openbanking-uk-rcs-api/src/main/java/com/forgerock/securebanking/openbanking/uk/rcs/mapper/decision/ConsentDecisionMapper.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.securebanking.openbanking.uk.rcs.mapper.decision;
+
+import com.forgerock.securebanking.openbanking.uk.rcs.api.dto.consent.decision.ConsentDecisionDeserialized;
+import com.forgerock.securebanking.platform.client.models.ConsentClientDecisionRequest;
+import lombok.extern.slf4j.Slf4j;
+import org.modelmapper.ModelMapper;
+import org.modelmapper.PropertyMap;
+import org.modelmapper.convention.MatchingStrategies;
+import org.springframework.stereotype.Component;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Class to map {@link ConsentDecisionDeserialized} to {@link ConsentClientDecisionRequest} to update the consent
+ */
+@Component
+@Slf4j
+public class ConsentDecisionMapper {
+
+    /**
+     * Map a {@link ConsentDecisionDeserialized} into an instance of type {@link ConsentClientDecisionRequest}
+     * @param consentDecisionDeserialized {@link ConsentDecisionDeserialized} to be map to {@link ConsentClientDecisionRequest}
+     * @return an instance of type {@link ConsentClientDecisionRequest}
+     */
+    public ConsentClientDecisionRequest map(ConsentDecisionDeserialized consentDecisionDeserialized) {
+        requireNonNull(
+                consentDecisionDeserialized,
+                "map(ConsentDecisionDeserialized) source parameter 'consentDecisionDeserialized' cannot be null"
+        );
+        ModelMapper modelMapper = mapper();
+        PropertyMap<ConsentDecisionDeserialized, ConsentClientDecisionRequest> decisionMap = new PropertyMap<>() {
+            protected void configure() {
+                map().getData().setStatus(source.getDecision());
+            }
+        };
+        // source, destination, map name
+        modelMapper.createTypeMap(ConsentDecisionDeserialized.class, ConsentClientDecisionRequest.class, getTypeMapName())
+                .addMapping(source -> source.getDebtorAccount().getFirstAccount(), ConsentClientDecisionRequest::setDataDebtorAccount)
+                .addMapping(source -> source.getConsentJwt(), ConsentClientDecisionRequest::setConsentJwt)
+                .addMappings(decisionMap);
```

Comment:
What is the benefit of using ModelMapper vs doing it manually? It seems to me that these are 2 fairly simple objects and that the resulting code is more verbose (and less efficient) than doing it by hand.

If we decide to stick with ModelMapper then check whether it is threadsafe. If it is, then it would make sense to create the mapper and configure the field mappings in the constructor for this class, this can then be reused per call to the mapping function.

Reply:
ModelMapper is threadsafe.

I did that implementation of ModelMapper before that refactor thinking that It could be a nice implementation to have a singleton instance of ModelMapper to map the different types of consents decisions to the same type of consent decision request object used to update the existing consent in IDM, but I now it's true that doesn't make sense for the simplicity of the implementation and because now the decision object it's a shared object.

I agree with you this code resulting in a verbose code and now doesn't give us any benefit, I will do some changes to implement that by hand and delete the model mapper.

Reply:
Done!